test/test_autograd.py

```diff
@@ -167,6 +167,26 @@ def backward(self, grad_output):
         MyFunction()(y).sum().backward()
         self.assertEqual(v.grad.data, torch.zeros(shape))
 
+    def test_legacy_function_deprecation_warning(self):
+        with warnings.catch_warnings(record=True) as w:
+            # Ensure warnings are being shown
+            warnings.simplefilter("always")
```

Comment:
Is this necessary for this warning to be shown? I'm not sure what the default warnings behavior is

Reply:
Oh, I see why you had to do this:
```
print the first occurrence of matching warnings for each location (module + line number) where the warning is issued
```